src/main/java/io/unlogged/core/bytecode/MethodVisitorWithoutProbe.java

```diff
@@ -0,0 +1,299 @@
+package io.unlogged.core.bytecode;
+
+import java.util.HashMap;
+
+import org.objectweb.asm.*;
+
+import io.unlogged.Constants;
+import io.unlogged.ExperimentalSDKFlagType;
+import io.unlogged.core.processor.UnloggedProcessor;
+import io.unlogged.util.ClassTypeUtil;
+
+class MethodVisitorWithoutProbe extends MethodVisitor {
+
+	private String methodName;
+	private String fullClassName;
+	private String desc;
+	private String nameProbed;
+	private long classCounter;
+	private long defaultCounter;
+	private ExperimentalSDKFlagType argumentSend;
+	private HashMap<String, Long> methodCounter = new HashMap<String, Long>();
+	private int access;
+
+	public MethodVisitorWithoutProbe(int api, String methodName, String fullClassName, int access, String desc, long classCounter, MethodVisitor mv) {
+		super(api, mv);
+		this.methodName = methodName;
+		this.fullClassName = fullClassName;
+		this.access = access;
+		this.desc = desc;
+		this.classCounter = classCounter;
+		this.nameProbed = this.methodName + Constants.probedValue;
+		this.defaultCounter = UnloggedProcessor.getDefaultCounter();
+		this.argumentSend = UnloggedProcessor.getArgumentSend();
```

Comment:
Multiple issues but related

This is a weird call to upstream. Why should `MethodVisitorWithoutProbe` know about the existence of UnloggedProcessor (a top level global) class. I think it shouldn't and instead the values for defaultCounter/argumentSend to be passed down. We have two values here. Wrap them in an new config class object to be passed down ?

Reply:
Yes, that would be a better data-flow scheme. I have implemented it now.